src/react-vite/12-testing/testing.md

```diff
@@ -38,71 +38,77 @@ React Testing Library’s north star is:
-rather than the implementation details of the components. This approach makes the tests more
-resilient to changes in the application’s codebase, as they are less likely to break with
-refactoring if the user interface and behavior remain consistent.
-
-> 2. It should be generally useful for testing the application components in the way the user would use it. We are making some trade-offs here because we’re using a computer and often a simulated browser environment, but in general, utilities should encourage tests that use the components the way they're intended to be used.
-
-The library promotes accessibility in React applications by providing tools and encouraging patterns
-that make it easier to create accessible web interfaces. By using semantic HTML and ARIA roles in
-tests, developers are nudged towards building applications that are more accessible to all users,
-including those using assistive technologies.
-
-> 3. Utility implementations and APIs should be simple and flexible.
-
-React Testing Library is designed to be lightweight and straightforward, avoiding the complexity that
-can sometimes come with extensive testing frameworks. It encourages good testing practices, such as
-avoiding testing the internal state of components and instead focusing on what the user would see
-and do. The library’s API is intentionally limited to encourage tests that align with these principles.
-
-### Rendering and verifying a component in a test
+rather than the implementation details of the components.
 
-Let’s take a look at some code that we added in [Handling User Inputs and
-Forms](./controlled-vs-uncontrolled.html).
+### <span id="rendering-and-verifying">Rendering and verifying a component in a test</span>
 
-@sourceref ../../../exercises/react-vite/11-controlled-vs-uncontrolled/03-problem/src/components/FormTextField/FormTextField.test.tsx
-@highlight 10-15, only
+OK, you're convinced that testing-library is a good idea, how do you use it to create tests for your
+React components?
 
-Each test consists of arguments to the Vite provided test function, `it`. The first argument is a
-short description of what the test is examining. The convention is that the description string takes
-"it" as a prefix and proceeds from there, e.g. "(it) renders with correct label and type."
+Let's take a look at an example; say we want to test a component we created named `FormTextField`
+that renders a form field. We expect that the component will generate HTML that looks like this:
 
-The second argument is a callback function that runs the test. In the example above the first line
-of the callback invokes `render` and passes it a single argument, JSX for the `FormTextField`
-component including props. After `render` completes the `screen` (a test DOM) will contain the
-elements created by our React code.
+```html
+<div>
+  <label htmlFor="inputId">Phone:</label>
+  <input id="inputId" type="tel" value="555-555-5555" />
+</div>
```

Comment:
> While simpler exercise code with less PMO code around it seems like a good idea, it might also be a "jarring" change from previous the context of previous examples that always built on PMO.

In the comment above, are you talking about how this is not referencing existing PMO code?

I think it’s _best_ if our example code in the explainers doesn’t use any PMO code, unless we are specifically trying to make a callback to PMO code. For example, see how all the examples here don’t have any overlap with PMO: https://www.bitovi.com/academy/learn-react/making-http-requests.html#objective-1-add-a-fetch-request-for-states

I think it’s better this way because then there’s less confusion with what they should put in their PMO code, and it makes people have to think about the concepts and then apply it in their PMO code (vs. just copy & paste).

So to that end, I would actually take this a step further and not use Phone as an example (because that’s in PMO). Also, I would use `<form>` instead of `<div>`. 😃

Reply:
I was referring to the possibility of the exercise code not using the PMO code.

The HTML example is the output of a single component, `FormTextField` not an entire form so it doesn't have a form element shown. The div contains the "field" component.

I will modify the example to not use direct PMO components but I'd like to keep using tel because it has useful characteristics in testing. There no tests for an input type of "tel" in any of the exercises.

Reply:
Changed to use email type to illustrate validation.